resources/plugincontent.py

```diff
@@ -111,20 +112,36 @@ def play_track(self):
                     "title":track['name'],
                     "genre": track["genre"],
                     "year": track["year"],
-                    "tracknumber": track["track_number"],
                     "album": track['album']["name"],
                     "artist": track["artist"],
                     "rating": track["rating"],
                     "duration": track["duration_ms"]/1000
                 }
+
+        if include_track_number:
+            infolabels["tracknumber"] = track["track_number"]
+
         li.setInfo( type="Music", infoLabels=infolabels)
         li.setProperty("spotifytrackid",track['id'])
         if KODI_VERSION > 15:
             li.setContentLookup(False)
         li.setProperty('do_not_analyze', 'true')
         
+        return url, li
+
+    def play_track(self):
+        track = self.sp.track(self.trackid, market=self.usercountry)
+        _url, li = self.parse_spotify_track(track)
         xbmcplugin.setResolvedUrl(handle=self.addon_handle, succeeded=True, listitem=li)
-        
+
+    def play_track_radio(self):
+        player = SpotifyRadioPlayer()
+        player.set_parent(self)
+        player.set_seed_track_id(self.trackid)
+        player.play()
+        while(not xbmc.abortRequested):
```

Comment:
Will take a look into this. I've also noticed an issue whereby the Spotify access token expires if left running too long (2 hours for me):

    File "/home/kodi/.kodi/addons/plugin.audio.spotify/resources/plugincontent.py", line 1258, in _fetch
    tracks = self._client.recommendations(seed_tracks=[t["id"] for t in self._buffer[0:5]], limit=self.FETCH_SIZE)["tracks"]
    File "/home/kodi/.kodi/addons/plugin.audio.spotify/resources/libs/spotipy/client.py", line 768, in recommendations
    return self._get('recommendations', **params)
    File "/home/kodi/.kodi/addons/plugin.audio.spotify/resources/libs/spotipy/client.py", line 134, in _get
    return self._internal_call('GET', url, payload, kwargs)
    File "/home/kodi/.kodi/addons/plugin.audio.spotify/resources/libs/spotipy/client.py", line 112, in _internal_call
    -1, '%s:\n %s' % (r.url, r.json()['error']['message']))
    SpotifyException: http status: 401, code:-1 - https://api.spotify.com/v1/recommendations?limit=20&seed_tracks=7JxEYE0xuJLf1GxppYL8Da%2C6u4v2Br0YijbtI0IhRkwkJ%2C0YdZ8DaTnj1sF2g2KMlS36%2C4XZmIqYNKCqZDaX8eVyUAZ%2C0PjwYHFrZSDU2OVIejBRkd:
    The access token expired

I suspect the answer to that problem may come from refactoring the above.